From e6bf1ba98fd396a1bc7c1872edf6fa98ac715ee7 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Mon, 5 Sep 2016 15:22:54 +0300
Subject: [PATCH 0471/1345] fix: mvpp2x: add cancel delayed work before
 destroy

commit  4c979d587fc5f99a0a899f4c3f45bc1e0fcff3d7 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- delayed work should be canceled before destroy
- fix crash during mvpp2x driver remove
- fix Jira SYSTEMSW-2517

Change-Id: I46b227d8e0577b979bdd654e1665786c89cc07cc
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32542
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 6429483..4a95eae 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -5157,6 +5157,7 @@ static int mv_pp2x_remove(struct platform_device *pdev)
 	struct mv_pp2x_hw *hw = &priv->hw;
 	int i;
 
+	cancel_delayed_work(&priv->stats_task);
 	flush_workqueue(priv->workqueue);
 	destroy_workqueue(priv->workqueue);
 
-- 
1.7.9.5

